data/battle_scripts_1.s

```diff
@@ -8386,6 +8586,18 @@ BattleScript_TerrainSeedLoop_NextBattler:
 	addbyte gBattlerTarget, 0x1
 	jumpifbytenotequal gBattlerTarget, gBattlersCount, BattleScript_TerrainSeedLoopIter
 	restoretarget
+	call BattleScript_ActivateSwitchInAbilities
```

Comment:
> is this a new gen 9 config thing?

Yes. Sort of? It's a change required for certain Gen. 9 abilities such as Protosynthesis and Quark Drive, as I explain in the description of the commit in which I added the `call` :eyes:

> if I'm not mistaken abilities like intimidate would trigger upon terrain seed consumption (and weather form changes above)

You're right. That happens 😔 

Maybe I should do something like this instead?

```diff
BattleScript_EffectMistyTerrain:
BattleScript_EffectGrassyTerrain:
BattleScript_EffectElectricTerrain:
BattleScript_EffectPsychicTerrain:
	attackcanceler
	attackstring
	ppreduce
	setterrain BattleScript_ButItFailed
	attackanimation
	waitanimation
	printfromtable gTerrainStringIds
	waitmessage B_WAIT_TIME_LONG
	playanimation BS_ATTACKER, B_ANIM_RESTORE_BG
	call BattleScript_TerrainSeedLoop
	jumpifabilitypresent ABILITY_MIMICRY, BattleScript_ApplyMimicry
+	jumpifterrainaffected BS_ATTACKER, STATUS_FIELD_ELECTRIC_TERRAIN, BattleScript_ElectricTerrainSwitchInAbilities
	goto BattleScript_MoveEnd

+BattleScript_ElectricTerrainSwitchInAbilities:
+	setbyte gBattlerTarget, 0
+BattleScript_MimicryLoopIter:
+	jumpifability BS_TARGET, ABILITY_QUARK_DRIVE, BattleScript_ElectricTerrainSwitchInAbilities_Activate
+BattleScript_MimicryLoop_NextBattler:
+	addbyte gBattlerTarget, 0x1
+	jumpifbytenotequal gBattlerTarget, gBattlersCount, BattleScript_MimicryLoopIter
+	goto BattleScript_MoveEnd

+BattleScript_ElectricTerrainSwitchInAbilities_Activate:
+	switchinabilities BS_TARGET
+	goto BattleScript_MimicryLoop_NextBattler
```

The deal here is that I need to activate the announcement of abilities like Protosynthesis and Quark Drive as soon as a sunny weather/electric terrain get set :eyes:

Reply:
Of course, in the case of Protosynthesis and the sunny weather, I added a `call BattleScript_ActivateSwitchInAbilities` in `BattleScript_MoveWeatherChange`. Do you think there's any edge case that could pop up there, @ghoulslash?

I could always apply a loop like the one I did for this example above though, I guess.

Reply:
to be honest I would just create a various/callnative e.g. `tryweatherchangeabilities` that checks specific ones. it would make it easy to add future abilities that activate on weather changes

Reply:
Hm. Wouldn't a new case in `AbilityBattleEffects` make more sense? For abilities triggered under specific weather conditions and another case for the terrains? :eyes: